PluginTemplateProject/src/main/scala/com/alpine/templates/db/TransformationTemplate.scala

```diff
@@ -0,0 +1,195 @@
+    val tableOrView = if (isView) "VIEW" else "TABLE"
+
+    //check if there is a table  or with the same name as the output table and drop according to the
+    // "overwrite"
+    if (overwrite) {
+      val stmtTable = connectionInfo.connection.createStatement()
+      //First see if a table of that name exists.
+      // This will throw an exception if there is a view with the output name,
+      // we will catch the exception and delete the view in the next block of code.
+      try {
+        listener.notifyMessage("Dropping table if it exists")
+        val dropTableStatementBuilder = new StringBuilder()
+        dropTableStatementBuilder ++= "DROP TABLE IF EXISTS " + fullOutputName + " CASCADE;"
+        stmtTable.execute(dropTableStatementBuilder.toString())
+      }
+      catch {
+        case (e: Exception) => listener.notifyMessage("A view of the name " + fullOutputName + "exists")
+      }
+      finally {
+        stmtTable.close()
+      }
+      //Now see if there is a view with the output name
+      listener.notifyMessage("Dropping view if it exists")
+      val dropViewStatementBuilder = new StringBuilder()
+      dropViewStatementBuilder ++= "DROP VIEW IF EXISTS " + fullOutputName + " CASCADE;"
+      val stmtView = connectionInfo.connection.createStatement()
+      stmtView.execute(dropViewStatementBuilder.toString())
+      stmtView.close()
+    }
+
+    //todo - pull out values from custom parameters
+
+
+    //todo - create the sql statement that will be executed using the parameters
+    val sqlStatement = s"""
+      CREATE $tableOrView $fullOutputName AS (
+        SELECT *
+        FROM ${getQuotedSchemaTableName(input.schemaName, input.tableName)}
+      );
+
+        """
+
+
+    //execute the SQL to create the new table
+    val stmt = connectionInfo.connection.createStatement()
+    stmt.execute(sqlStatement)
+    stmt.close()
+
+    //todo - create the output schema (set to input schema for now)
+    val outputTabularSchema = input.tabularSchema
+
+    //
+    //metadata about output table.  At runtime, next operator uses this to know where the data is.  Also passed to visualization function. However, default works so not overriding it.
+    /**
+     *  returns implementation of DBTable.  Used in two ways:
+     *  (1) At runtime, the following operator uses this information to know where its input data is coming from
+     *  (2) At runtime, this object is passed to the visualization function (not applicable here as we use the default visualization for a DBTable)
+
+     */
+
+    DBTableDefault(
+      outputSchema,
+      outputName,
+      outputTabularSchema,
+      isView,
+      connectionInfo.name,
+      connectionInfo.url,
+      Some(params.operatorInfo)
+    )
+  }
+
+  //helper functions.  database tables that are not lowercase need to be put into quotes to be read properly
+  def getQuotedSchemaTableName(schemaName: String, tableName: String): String = {
```

Comment:
I think you're right; everything in here is specific for GP/PG (like all the drop table/view code).  I can put in comments perhaps at the beginning of the onExecution indicating its specified database support?

Once the framework knows how to handle multiple data sources, then we can redo the template and make it database agnostic.